pkg/resource/api.go

```diff
@@ -154,6 +154,10 @@ func NewAPIManagedBinder(c client.Client) *APIManagedBinder {
 // Bind the supplied resource to the supplied claim.
 func (a *APIManagedBinder) Bind(ctx context.Context, cm Claim, mg Managed) error {
 	cm.SetBindingPhase(v1alpha1.BindingPhaseBound)
+	// Propagate back the final name of the external resource to the claim.
+	if meta.GetExternalName(mg) != "" {
```

Comment:
How likely is it that the external name annotation will be unset at binding time? Should we ensure the managed resource does not become bindable until its external name is set?

Reply:
> How likely is it that the external name annotation will be unset at binding time?

Only if developer of a resource, which acquires its name from the provider, forgets to add it as annotation & does not use the annotation in the provider queries. Or there is a bug and external name is set and then unset for some reason. I don't know, not very likely given that in all queries they'd depend on this annotation.

> Should we ensure the managed resource does not become bindable until its external name is set?

It's merely informational on `Claim` level after its creation, so, not sure if we want it to block the binding process. But one can also say, we'd have everyone onboard to use it if we don't bind it otherwise. Although, this would break all controllers who do not use the annotation until they do.

Reply:
> Although, this would break all controllers who do not use the annotation until they do.

That's a good point - we have plenty of controllers that don't yet use the managed resource reconciler (and thus don't honor this external name annotation) and probably won't for some time. Given that all resources _do_ use this resource claim reconciler I agree we shouldn't assume the annotation is present.